api/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -37,10 +37,18 @@ type OpenTelemetryCollectorSpec struct {
 	Replicas *int32 `json:"replicas,omitempty"`
 
 	// Image indicates the container image to use for the OpenTelemetry Collector.
+	// Deprecated. Use a DistributionName pointing to a distribution with a custom image instead. This field will be be noop in v0.16 and removed in v0.17.
+	// Cannot be specified when a DistributionName is set.
 	// +optional
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	Image string `json:"image,omitempty"`
 
+	// Distribution specifies which distribution to use for this instance. In case the distribution does not exist, this resource will wait until it is created.
```

Comment:
Great question. I think we should prevent distributions from being removed, via webhook ( see `ValidateDelete` for `OpenTelemetryCollector`). If the webhook is disabled for some reason, we'd still have cases where distributions are removed when a otelcol still references them. In which case, I'd make it work as if the distribution isn't there (ie, fail the reconciliation with a reschedule).

> Also, regarding "In case the distribution does not exist, this resource will wait until it is created." - I didn't see any logic related to this?

You are right, I forgot about this logic. I'm now glad I left a comment with what I had in mind when designing this :-)